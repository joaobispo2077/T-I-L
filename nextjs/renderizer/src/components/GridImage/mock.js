export default {
	hasBackground: false,
	title: 'My grid',
	description:
		'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
	grid: [
		{
			altText: 'Teste 1',
			src: 'https://source.unsplash.com/random/800x800?r=1',
		},
		{
			altText: 'Teste 2',
			src: 'https://source.unsplash.com/random/800x800?r=2',
		},
		{
			altText: 'Teste 3',
			src: 'https://source.unsplash.com/random/800x800?r=3',
		},
		{
			altText: 'Teste 4',
			src: 'https://source.unsplash.com/random/800x800?r=4',
		},
		{
			altText: 'Teste 5',
			src: 'https://source.unsplash.com/random/800x800?r=5',
		},
		{
			altText: 'Teste 6',
			src: 'https://source.unsplash.com/random/800x800?r=6',
		},
	],
};
