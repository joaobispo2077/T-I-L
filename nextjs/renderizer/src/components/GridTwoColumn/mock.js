export default {
  title: 'GridTwoColumn',
  description:
    'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Suscipit quisquam aspernatur sed quos, dolore cum eveniet magnam at consequatur totam iste minus, blanditiis obcaecati eos rerum sunt dolorum perferendis. Nam?',
  image: {
    url: 'assets/images/javascript.svg',
    alternativeText: 'bla bla',
  },
  background: true,
};
