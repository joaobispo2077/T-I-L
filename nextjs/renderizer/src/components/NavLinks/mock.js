export default [
	{
		id: 1,
		text: 'Link 1',
		link: '#target1',
		newTab: false,
	},
	{
		id: 2,
		text: 'Link 2',
		link: '#target2',
		newTab: false,
	},
	{
		id: 3,
		text: 'Link 3',
		link: '#target3',
		newTab: true,
	},
	{
		id: 4,
		text: 'Link 4',
		link: '#target4',
		newTab: false,
	},
	{
		id: 5,
		text: 'Link 5',
		link: '#target5',
		newTab: false,
	},
];
