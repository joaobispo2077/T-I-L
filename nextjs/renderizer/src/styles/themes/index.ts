export const themes = {
  default: {
    colors: {
      primary: '#0A1128',
      secondary: '#dc143C',
      light: '#FFFFFF',
      dark: '#0A1128',
      mediumGray: '#DDDDDD',
    },
    font: {
      family: {
        primary: "'Open Sans', sans-serif",
        secondary: "'Montserrat', sans-serif",
      },
      sizes: {
        xsmall: '.8rem',
        small: '1.6rem',
        medium: '2.4rem',
        large: '3.2rem',
        xlarge: '4.0rem',
        xxlarge: '4.8rem',
        huge: '5.6rem',
        xhuge: '6.4rem',
      },
    },
    spacings: {
      xsmall: '.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
    media: {
      lteMedium: '(max-width: 768px)',
    },
  },
} as const;
