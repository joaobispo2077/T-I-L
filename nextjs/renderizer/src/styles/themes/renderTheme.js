import { ThemeProvider } from 'styled-components';
import { themes } from '.';

export const renderTheme = (children) => {
	return (
		<ThemeProvider theme={{ ...themes['default'] }}>{children}</ThemeProvider>
	);
};
