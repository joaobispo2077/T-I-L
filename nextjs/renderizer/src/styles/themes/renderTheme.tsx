import React from 'react';
import { ThemeProvider } from 'styled-components';
import { themes } from '.';

export const renderTheme = (children: React.ReactNode) => {
  return (
    <ThemeProvider theme={{ ...themes.default }}>{children}</ThemeProvider>
  );
};
