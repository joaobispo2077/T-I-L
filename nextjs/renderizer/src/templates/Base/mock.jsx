import linksMock from '../../components/NavLinks/mock';

import { GridText } from '../../components/GridText';
import gridMock from '../../components/GridText/mock';

export default {
	links: linksMock,
	logo: {
		text: 'Logo',
		link: '#',
	},
	footer: '<p>testando footer</p>',
	children: (
		<div>
			<GridText {...gridMock} hasBackground />
			<GridText {...gridMock} />
			<GridText {...gridMock} hasBackground />
			<GridText {...gridMock} />
		</div>
	),
};
