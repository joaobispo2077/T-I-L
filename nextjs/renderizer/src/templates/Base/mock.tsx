import linksMock from '../../components/NavLinks/mock';

import { GridText } from '../../components/GridText';
import gridMock from '../../components/GridText/mock';
import { BaseProps } from '.';

export default {
  links: linksMock,
  logo: {
    text: 'Logo',
    link: '#',
  },
  footer: '<p>testando footer</p>',
  children: (
    <div>
      <GridText {...gridMock} hasBackground />
      <GridText {...gridMock} />
      <GridText {...gridMock} hasBackground />
      <GridText {...gridMock} />
    </div>
  ),
} as BaseProps;
