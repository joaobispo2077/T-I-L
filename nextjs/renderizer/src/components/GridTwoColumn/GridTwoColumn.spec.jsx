import { render } from '@testing-library/react';
import { GridTwoColumn } from '.';
import { renderTheme } from '../../styles/themes/renderTheme';
import mock from './mock';

describe('<GridTwoColumn/>', () => {
	it('should match with snapshot', () => {
		const { container } = render(renderTheme(<GridTwoColumn {...mock} />));

		expect(container).toMatchSnapshot();
	});
});
