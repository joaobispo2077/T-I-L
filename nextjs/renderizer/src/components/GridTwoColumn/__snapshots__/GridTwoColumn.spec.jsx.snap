// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn/> should match with snapshot 1`] = `
.c5 {
  color: #FFFFFF;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #0A1128;
  color: #FFFFFF;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
    text-align: center;
  }
}

<div>
  <section
    class="c0"
  >
    <section
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            GridTwoColumn
          </h2>
          <footer
            class="c6"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Suscipit quisquam aspernatur sed quos, dolore cum eveniet magnam at consequatur totam iste minus, blanditiis obcaecati eos rerum sunt dolorum perferendis. Nam?
          </footer>
        </div>
        <div
          class=""
        >
          <img
            alt="GridTwoColumn"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </section>
  </section>
</div>
`;
